### PR TITLE
Backwards incompatible changes for 4.0

### PR DESCRIPTION
**Pull request by oliver006**
Couple of breaking changes that will go into 4.0

In particular, this one addresses PRs:
- #174 
- #126 
- #106 
- #37 
- #89

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,15 +133,15 @@ of the API. If you do not use this option you will be defaulted to the oldest st
 client := goshopify.NewClient(app, "shopname", "", goshopify.WithVersion("2019-04"))
 ```
 
-#### WithRetry
+#### WithRetry, err
```
Comment (BoldBrandonM) [inline README.md:136]: This line can be reverted

File: abandoned_checkout.go [hunk 2]
```diff
@@ -47,10 +47,10 @@ type AbandonedCheckout struct {
 	Currency                 string               `json:"currency,omitempty"`
 	CompletedAt              *time.Time           `json:"completed_at,omitempty"`
 	ClosedAt                 *time.Time           `json:"closed_at,omitempty"`
-	UserID                   int64                `json:"user_id,omitempty"`
+	UserId                   uint64               `json:"user_id,omitempty"`
```
Comment (BoldBrandonM) [inline abandoned_checkout.go:50]: Apologies if it's readily apparent elsewhere, but what prompted the change to uint64?

File: goshopify.go [hunk 3]
```diff
@@ -244,21 +244,33 @@ func (c *Client) NewRequest(method, relPath string, body, options interface{}) (
...
+func (app App) NewClient(shopName, token string, opts ...Option) (*Client, error) {
 	return NewClient(app, shopName, token, opts...)
 }
 
 // Returns a new Shopify API client with an already authenticated shopname and
 // token. The shopName parameter is the shop's myshopify domain,
 // e.g. "theshop.myshopify.com", or simply "theshop"
-func NewClient(app App, shopName, token string, opts ...Option) *Client {
+// panics if an error occurs
+func MustNewClient(app App, shopName, token string, opts ...Option) *Client {
```
Comment (BoldBrandonM) [inline goshopify.go:259]: @I'm not sold that we want to expose a possible panic from this library even if it's heavily documented. Is there a strong rationale as to why someone wouldn't prefer to do this in their client code? I'd say lets just remove the panic altogether, but that's certainly an opinion.

**Review by BoldBrandonM — CHANGES_REQUESTED** [review]
The only thing here I see that I'm not keen on is the constructor panic. Looks great otherwise!

File: abandoned_checkout.go (hunk 2 above)
Comment (oliver006) [reply abandoned_checkout.go:51]: https://github.com/bold-commerce/go-shopify/issues/106

Comment (oliver006) [reply abandoned_checkout.go:51]: Changed all IDs to uint64

File: goshopify.go (hunk 3 above)
Comment (oliver006) [reply goshopify.go:262]: For instance the prometheus library uses MustRegister() which will panic if the call fails so this is not without precedence in the golang ecosystem.

I dislike the panic and also the return error. It's so unnecessary. All that is checked inside the NewClient() call is the url.Parse() which is silly. It just doesn't seem worth having the error returned and checked for that. 

I kept the panic as MustNewClient() so it's a drop-in replacement for the current NewClient() - I don't feel strongly about keeping it but for instance in my code I would just call NewClient() and ignore the error - I know that the domain is good cause I take whatever comes from shopify and if it's not then 🤷‍♀️   - the call will fail later.

File: README.md (hunk 1 above)
Comment (oliver006) [reply README.md:136]: Good catch, that should have gone in the line above, for the call to Newclient().
Fixed.

**Review by BoldBrandonM — APPROVED** [review]

**Review by BoldBrandonM — COMMENTED** [review]
Looks like some changes to customer and order APIs were pulled in, are they related to the breaking changes?

**Comment by oliver006** [comment]
> Looks like some changes to customer and order APIs were pulled in, are they related to the breaking changes?

I did some merging - rebasing after the recent PRs #260 and #261 and #259 which modified order.go and customer.go but I just looked at the diff here and it seems fine. Anything sticking out that shouldn't be there?

**Comment by BoldBrandonM** [comment]
> > Looks like some changes to customer and order APIs were pulled in, are they related to the breaking changes?
> 
> I did some merging - rebasing after the recent PRs #260 and #261 and #259 which modified order.go and customer.go but I just looked at the diff here and it seems fine. Anything sticking out that shouldn't be there?

Oh I was just looking at the diff between the pushes, but if that's just from the master branch then no worries.

**Review by BoldBrandonM — APPROVED** [review]
Reapproving, as this latest change was only master rebases.

**Pull request merged**